iPetCare/Application/Services/RaceService.cs

```diff
@@ -0,0 +1,182 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Application.Dtos.Races;
+using Application.Interfaces;
+using Application.Services.Utilities;
+using AutoMapper;
+using Domain.Models;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using Persistence;
+
+namespace Application.Services
+{
+    public class RaceService : IRaceService
+    {
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly DataContext _context;
+        private readonly IUserAccessor _userAccessor;
+        private readonly IMapper _mapper;
+
+        public RaceService(UserManager<ApplicationUser> userManager, DataContext context, IUserAccessor userAccessor, IMapper mapper)
+        {
+            _context = context;
+            _userAccessor = userAccessor;
+            _userManager = userManager;
+            _mapper = mapper;
+        }
+        public async Task<ServiceResponse<RaceCreateDtoResponse>> CreateAsync(RaceCreateDtoRequest dto)
+        {
+            if (await _context.Races.Where(x => x.Name == dto.Name).AnyAsync())
+                return new ServiceResponse<RaceCreateDtoResponse>(HttpStatusCode.BadRequest, "Podana rasa już istnieje");
+
+            var currentUserName = _userAccessor.GetCurrentUsername();
+
+            if (currentUserName != null)
+            {
+                var currentUser = await _userManager.FindByNameAsync(currentUserName);
+                if (currentUser != null && currentUser.Role != Role.Administrator)
```

Comment:
Czy tu nie trzeba dodać "|| currentUser == null"?

Reply:
W zasadzie to chyba faktycznie powinniśmy sprawdzić czy user jest nullem jak @Pawel-Kot  pisze. Tylko że wtedy już nie forbidden ale unauthorized. To sprawdzenie dodałbym do poniższego ifa z currentUserName == null (linijka 44)